src/goes/goes_util.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+
+
+#
+# goes_util.py
+#
+# This class contains methods for sub-sampling data arrays and filtering data arrays. The
+# sub-sampling techniques used in this class are derived from section 3.4.3 in the
+# "GOES-R Advanced Baseline Imager (ABI) Algorithm Theoretical Basis Document For Cloud
+# and Moisture Imagery Product (CMIP)" Version 3.0 July 30, 2012
+# (https://www.star.nesdis.noaa.gov/goesr/docs/ATBD/Imagery.pdf).
+#
+class GoesUtil:
+
+    def __init__(self):
+        """
+        Constructor
+        """
+        self._increment = None
+        self._yaw_flip_flag = None
+        self._resolution = None
+        self._fill_value_index_array = None
+
+    def set_yaw_flip_flag(self, yaw_flip_flag):
+        """
+        Sets the yaw_flip_flag variable
+        yaw_flip_flag - the yaw_flip_flag
+        """
+        self._yaw_flip_flag = yaw_flip_flag
+
+    def set_resolution(self, resolution):
+        """
+        Sets the resolution variable and array slicing increment for sub_sampling
+        resolution - the resolution
+        """
+        self._resolution = resolution
+        self._increment = int(self._resolution / 2)
+
+    def set_fill_value_index_array(self, fill_value_index_array):
+        """
+        Sets the fill_value_index_array
+        fill_value_index_array - an array of indices to remove
+        """
+        self._fill_value_index_array = fill_value_index_array
```

Comment:
Comment: probably good to verify the usage of fill value here, and that it's consistent with code expectations if you haven't done so already.

Reply:
I will add more comments in the code regarding this.